charts/images.yaml

```diff
@@ -11,6 +11,10 @@ images:
   sourceRepository: github.com/gardener/machine-controller-manager
   repository: eu.gcr.io/gardener-project/gardener/machine-controller-manager
   tag: "v0.38.0"
+- name: machine-controller-manager-provider-equinix-metal
+  sourceRepository: github.com/gardener/machine-controller-manager-provider-equinix-metal
+  repository: eu.gcr.io/gardener-project/gardener/machine-controller-manager-provider-equinix-metal
+  tag: "v0.1.0"
```

Comment:
/hold until https://github.com/gardener/machine-controller-manager-provider-equinix-metal is released.

In the meanwhile `eu.gcr.io/gardener-project/gardener/machine-controller-manager-provider-equinix-metal:v0.1.0-dev-c2fcae3e2bc087cc34e6e3a21c8e56380f800210` can be used for testing. Once tests looks good, we can trigger the release button for it.

Reply:
Good point. Is there anything holding us up from merging the last PR for https://github.com/gardener/machine-controller-manager-provider-equinix-metal and releasing it?

Reply:
No. I shall go ahead and release the MCM provider Equinix metal then?

Reply:
I am good to go with that one.

Reply:
[machine-controller-manager-provider-equinix-metal:v0.1.0](https://github.com/gardener/machine-controller-manager-provider-equinix-metal/releases/tag/v0.1.0) is now ready. 

/unhold

Reply:
Is the CI on this stuck?

Reply:
There are failures for the unit-tests here. Might need to fix them.

```
[Fail] Machines workerDelegate #GenerateMachineDeployments, #DeployMachineClasses machine images [It] should delete the all old PacketMachineClasses 
/go/src/github.com/gardener/gardener-extension-provider-packet/vendor/github.com/gardener/gardener/pkg/mock/controller-runtime/client/mocks.go:101

[Fail] Machines workerDelegate #GenerateMachineDeployments, #DeployMachineClasses machine images [It] should return err when the infrastructure provider status cannot be decoded 
/go/src/github.com/gardener/gardener-extension-provider-packet/vendor/github.com/gardener/gardener/pkg/mock/controller-runtime/client/mocks.go:101

[Fail] Machines workerDelegate #GenerateMachineDeployments, #DeployMachineClasses machine images [It] should not delete the any of old PacketMachineClasses as DeleteAll call returns error 
/go/src/github.com/gardener/gardener-extension-provider-packet/vendor/github.com/gardener/gardener/pkg/mock/controller-runtime/client/mocks.go:101

Ran 11 of 11 Specs in 0.019 seconds
FAIL! -- 8 Passed | 3 Failed | 0 Pending | 0 Skipped
```

Reply:
I don't know what these mean. How do I reproduce them?

Reply:
The unit-tests can be invoked [from here](https://github.com/gardener/gardener-extension-provider-packet/blob/master/Makefile#L113). 
```
make test
```

Reply:
@prashanth26 I managed to figure most of it (was quite painful to get there). It now has 2 tests failing, both because it expects to get an error when passing an invalid infrastructure provider status as `w.worker.Spec.InfrastructureProviderStatus.Raw`, yet it does not. [This line](https://github.com/gardener/gardener-extension-provider-packet/pull/130/files#diff-ede719e7797a6dd04b91acb4296b78434730159686b71e616b1c989712a7716eR102) always passes without error:

```go
	if _, _, err := w.Decoder().Decode(w.worker.Spec.InfrastructureProviderStatus.Raw, nil, infrastructureStatus); err != nil {
		return err
	}
```

Although the original it was [based upon](https://github.com/gardener/gardener-extension-provider-aws/blob/4b402bd01068b6bbd0a31b188bc2d16cd2b91d19/pkg/controller/worker/machines.go#L88-L91) has the same behaviour, yet passes:

```go
	infrastructureStatus := &awsapi.InfrastructureStatus{}
	if _, _, err := w.Decoder().Decode(w.worker.Spec.InfrastructureProviderStatus.Raw, nil, infrastructureStatus); err != nil {
		return err
	}
```

What should be causing it to fail, and why is it not being triggered?

Reply:
For the test case `should fail because the infrastructure status cannot be decoded`,

The behaviour on your test case seems to be right. An empty infra status decode should ideally return a decoded object with empty/blank values. So Ideally i think the test case shouldn't fail. So you can update your test case to not return an error after fixing the mocking errors by initialization secret checks and DeleteAllMocks. Or you can also get rid of this test if it doesn't make sense. 

I verified this with the `provider-aws` test as the test seems to be failing elsewhere with this message `no instance profile with purpose "nodes" found`. Will fix this.

<img width="1197" alt="Screenshot 2021-04-14 at 10 08 13 AM" src="https://user-images.githubusercontent.com/34744389/114656125-253eb600-9d0b-11eb-81cf-5a504f5e0a58.png">

---

With the other test case failing `should return err when the infrastructure provider status cannot be decoded`,

The test case fails as mocking function for secret is setup but never invoked. Please remove this line `expectGetSecretCallToWork(c, packetAPIToken, packetProjectID)` and the test should succeed.

Reply:
> should return err when the infrastructure provider status cannot be decoded

Based on your explanation, I see no reason for the test. Removed.

Reply:
> With the other test case failing should return err when the infrastructure provider status cannot be decoded,
>
> The test case fails as mocking function for secret is setup but never invoked. Please remove this line expectGetSecretCallToWork(c, packetAPIToken, packetProjectID) and the test should succeed.

I don't know what you mean. The entire test case is this:

```go
				It("should return err when the infrastructure provider status cannot be decoded", func() {
					// Deliberately setting InfrastructureProviderStatus to empty
					w.Spec.InfrastructureProviderStatus = &runtime.RawExtension{}
					workerDelegate, _ := NewWorkerDelegate(common.NewClientContext(c, scheme, decoder), chartApplier, "", w, cluster)

					err := workerDelegate.DeployMachineClasses(context.TODO())
					Expect(err).To(HaveOccurred())
				})
```

Where do you see in it `expectGetSecretCallToWork()`?

Reply:
Ah, never mind, the names were reversed in the comment. I see it now.

Reply:
Done